### DISCUSSION TITLE
The scaling is off? Can we get a tab\tool\workflow to create matching spectrograms?

### DISCUSSION THREAD

**Discussion by TheRealBlackNet** [riffusion/riffusion#112] [Ideas]
![grafik](https://user-images.githubusercontent.com/115030801/216537546-6a2c1efd-49a1-42b0-a779-bcb25125cbac.png)
We see left the input image with the Herz and the pixels from top.
Left this is a linear spectrogram by "Sonic Visualiser".
**Observation:**
Looks like some bands are more detailed (have more pixels) then others.

**Background:**
I wanted to use the Automatic1111 riffusion tab to create sounds, but not AI generated they should be placed by my own tool. 
Idea is I have a buch of stencils of music instruments and I copy them into a image on the timeframe the input will be a Description from outside like:
Flute: |---x---|-x---x-| ...
Harp: |-x--x--|--x-x--| ...
With each x I copy the stencil into the image with n * beat speed the distance between. (Its a Dwarf Fortress music description)

**Problem:**
My first tests with Paint sounded off so I started to check the frequencies that leads to the image above. Moving the pixels from a specrogram by hand sounds not nice in the reconstruction.
I tired to use the model but getting only a single sound from a flute, piano and so on is not possible a "just a C on a flute" is not trained from the feeling.

**Question\Featurerequest\Idea:**
Can we please get a "input wav receive a useable spectogramm" tool on the web app or in the Automatic1111 tab?
So I can create a set of correct scaled stencils from the spectogram and create a collage of sounds.

**Comment by johndpope** [comment]
sounds reasonable. 

I saw this popup - https://civitai.com/?query=riffusion
With advances in stable diffusion  SDXL - 1024 x 1024 - https://twitter.com/EMostaque/status/1649839142073671682
hopefully this helps this space.

this may also help
https://github.com/enlyth/sd-webui-riffusion

(looking at the flute / harp requirement - you may look into band in a box )
 seems like we're a year away from finer control over audio. 

UPDATE - just found this
https://github.com/johndpope/riff-cnet
